python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -783,9 +783,13 @@ def build_run_requests(
                 check.failed("Partition key provided for unpartitioned asset")
             tags.update({**partitions_def.get_tags_for_partition_key(partition_key)})
 
+        # Normally the run request would have to be resolved with the partitions def, but
```

Comment:
I find this comment a little confusing - it seems like it's trying to explain the context for why we're doing something weird in the code, but the code below looks perfectly expected.

Is this explaining why we have:
```python
        if partition_key is not None:
            if partitions_def is None:
                check.failed("Partition key provided for unpartitioned asset")
            tags.update({**partitions_def.get_tags_for_partition_key(partition_key)})
```

Reply:
Ah the comment is referring to how we don't do `run_request.with_resolved_partition(...)` like we do with all other run requests with a partition key.

`with_resolved_partition` is responsible for confirming that the partition exists and applying the necessary tags to the run request, but this causes the asset reconciliation sensor to perform significantly slower, so we directly instantiate here instead.

I'll update this comment to clarify

Reply:
How do we avoid doing `run_request.with_resolved_partition`? Doesn't it end up getting called by the sensor machinery once this function returns?

Reply:
we check if run requests have resolved partitioning via `run_request.has_resolved_partition_run_request()`, which checks if the partition tag exists. This allows run requests generated here and via `run_request_for_partition` to bypass the `with_resolved_partition` function.

If we didn't add that, existing user tests may fail when `repository_def` isn't provided to the sensor/schedule context, since the sensor will assume that run requests created via `run_request_for_partition` require resolution and thus will try to load the repository.